internal/cmd/workspaces.go

```diff
@@ -752,3 +753,76 @@ func buildUpdateReq(ctx context.Context, client coder.Client, conf updateConf) (
 	}
 	return &updateReq, nil
 }
+
+func setPolicyTemplate() *cobra.Command {
+	var (
+		ref      string
+		repo     string
+		filepath string
+		dryRun   bool
+	)
+
+	cmd := &cobra.Command{
+		Use:   "policy-template",
+		Short: "Set workspace policy template",
+		Long:  "Set workspace policy template",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return err
+			}
+
+			if filepath == "" && ref == "" && repo == "" {
+				return clog.Error("Must specify a configuration source",
+					"A template source is either sourced from a local file (-f) or from a git repository (--repo-url and --ref)",
+				)
+			}
+
+			var rd io.Reader
+			if filepath != "" {
+				b, err := ioutil.ReadFile(filepath)
+				if err != nil {
+					return xerrors.Errorf("read local file: %w", err)
+				}
+				rd = bytes.NewReader(b)
+			}
+
+			req := coder.ParseTemplateRequest{
+				RepoURL:  repo,
+				Ref:      ref,
+				Local:    rd,
+				OrgID:    coder.SkipTemplateOrg,
+				Filepath: ".coder/coder.yaml",
+			}
+
+			version, err := client.ParseTemplate(ctx, req)
+			if err != nil {
+				return handleAPIError(err)
+			}
+
+			resp, err := client.SetPolicyTemplate(ctx, version.TemplateID, coder.TemplateScopeSite, dryRun)
+			if err != nil {
+				return handleAPIError(err)
+			}
+
+			for _, mc := range resp.MergeConflicts {
+				workspace, err := client.WorkspaceByID(ctx, mc.WorkspaceID)
```

Comment:
I like the extra details. In my dogfood of 200 workspaces this was still pretty quick :+1: